client-registry-jempi/swarm.sh

```diff
@@ -1,156 +1,178 @@
 #!/bin/bash
 
-readonly ACTION=$1
-readonly MODE=$2
-
-COMPOSE_FILE_PATH=$(
-  cd "$(dirname "${BASH_SOURCE[0]}")" || exit
-  pwd -P
-)
-readonly COMPOSE_FILE_PATH
-
-ROOT_PATH="${COMPOSE_FILE_PATH}/.."
-readonly ROOT_PATH
-
-. "${ROOT_PATH}/utils/config-utils.sh"
-. "${ROOT_PATH}/utils/docker-utils.sh"
-. "${ROOT_PATH}/utils/log.sh"
-
-service_names=(
-  "jempi-kafka-01"
-  "jempi-kafka-02"
-  "jempi-kafka-03"
-  "jempi-kafdrop"
-  "jempi-zero-01"
-  "jempi-alpha-01"
-  "jempi-alpha-02"
-  "jempi-alpha-03"
-  "jempi-ratel"
-  "jempi-async-receiver"
-  "jempi-sync-receiver"
-  "jempi-pre-processor"
-  "jempi-controller"
-  "jempi-em-calculator"
-  "jempi-linker"
-  "jempi-api"
-)
-readonly service_names
+declare ACTION=""
+declare MODE=""
+declare COMPOSE_FILE_PATH=""
+declare UTILS_PATH=""
+declare KAFKA_SERVICES=()
+declare DGRAPH_SERVICES=()
+declare COMBINED_SERVICES=()
+declare SERVICE_NAMES=()
+declare VOLUME_NAMES=()
+
+function init_vars() {
+  ACTION=$1
+  MODE=$2
+
+  COMPOSE_FILE_PATH=$(
+    cd "$(dirname "${BASH_SOURCE[0]}")" || exit
+    pwd -P
+  )
+
+  UTILS_PATH="${COMPOSE_FILE_PATH}/../utils"
+
+  DGRAPH_SERVICES=("jempi-ratel")
+  VOLUME_NAMES=("jempi-zero-01-data")
```

Comment:
jempi-zero-01-data will be duplicated.

Reply:
can you check again? the loop is for kafka and alpha and not zero volumes